File: Programming Skills II/896. Monotonic Array/896. Monotonic Array.py
class Solution(object):
    def isMonotonic(self, nums):
        dif = 0
        for i in range( len(nums) -1):
            if nums[i] != nums[i+1] and dif == 0 :
                dif = nums[i] - nums[i+1]
            elif (nums[i] > nums[i+1] and dif < 0)  or (nums[i] < nums[i+1] and dif > 0) :
                return False
        return True

File: Programming Skills II/896. Monotonic Array/896. Monotonic Array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        int dif = 0;
        for(int i=0; i<nums.length-1; ++i){
            if(nums[i]!=nums[i+1] && dif == 0)
                dif = nums[i] - nums[i+1];
            else if( ( nums[i] > nums[i+1] && dif < 0 ) || (nums[i] < nums[i+1] && dif > 0) )
                return false;
        }
        return true;
    }
}
